proxy/src/config.rs

```diff
@@ -185,7 +185,13 @@ impl<'a> TryFrom<&'a Strings> for Config {
             parse(strings, ENV_METRICS_FLUSH_INTERVAL_SECS, parse_number);
         let report_timeout = parse(strings, ENV_REPORT_TIMEOUT_SECS, parse_number);
         let pod_name = strings.get(ENV_POD_NAME);
-        let pod_namespace = strings.get(ENV_POD_NAMESPACE);
+        let pod_namespace = strings.get(ENV_POD_NAMESPACE).and_then(|maybe_value| {
+            // There cannot be a default pod namespace, and the pod namespace is required.
+            maybe_value.ok_or_else(|| {
+                error!("{} is not set", ENV_POD_NAMESPACE);
```

Comment:
not a blocker, but it seems cleaner for this function to return an error type that can be logged by the caller. our current error type seems insufficient for a meaningful log message.

Reply:
> not a blocker, but it seems cleaner for this function to return an error type that can be logged by the caller. our current error type seems insufficient for a meaningful log message.

The error type is intentionally vague because the logging actually happens in Config.rs already, so the caller doesn't need to log anything. In particular, the `error!()` here and `strings.get()` will log the errors. Note that every error for every configuration parameter is logged before the function returns; in particular, if multiple parameters have errors, then all those errors are logged before we give up.